internal/chain/engine.go

```diff
@@ -254,9 +254,28 @@ func (e *engineImpl) Finalize(
 					totalWithdraw := delegation.RemoveUnlockedUndelegations(header.Epoch())
 					state.AddBalance(delegation.DelegatorAddress, totalWithdraw)
 				}
+
+				stats, err := chain.ReadValidatorStats(validator)
+				if err != nil {
+					return nil, nil, ctxerror.New("could not read validator stats").WithCause(err)
+				}
+
+				count := len(stats.Delinquency.Missing)
+				didMiss := 0
+
+				for _, mia := range stats.Delinquency.Missing {
+					if mia.Count >= slash.MissedThresholdForInactive {
+						didMiss++
+					}
+				}
+				if didMiss >= ((count * (2 / 3)) + 1) {
```

Comment:
This is only counting 66% of nodes that' not signing for the whole epoch. I was saying counting 66% of all the blocks that the validator should sign but missed across all keys.